From fc3a0a609fcf31886d047efb03cb84068fe4fac9 Mon Sep 17 00:00:00 2001
From: Phil Dennis <pjdennis@gmail.com>
Date: Tue, 10 Feb 2026 07:28:50 -0600
Subject: [PATCH 12/16] Remove obsolete process_conditional_directive

Delete process_conditional_directive function (31 lines) which is no longer needed with unified parsing. Conditional directives are now handled through the normal process_directive dispatch with SKIP_DEPTH guards. Update comments in asm.asm and directives.asm.
---
 assembler2/23/asm.asm        |  2 +-
 assembler2/23/directives.asm | 39 +++---------------------------------
 2 files changed, 4 insertions(+), 37 deletions(-)

diff --git a/assembler2/23/asm.asm b/assembler2/23/asm.asm
index fa7839f..49a9eb0 100644
--- a/assembler2/23/asm.asm
+++ b/assembler2/23/asm.asm
@@ -70,7 +70,7 @@
 ;   read_char (asm.asm alias; implemented in file_stack.asm)
 ;   read_token, skip_token, skip_rest_of_line, check_for_end_of_line (tokenizer.asm)
 ;   capture_label (labels.asm)
-;   process_directive, process_conditional_directive (directives.asm)
+;   process_directive (directives.asm)
 ;   lookup_mnemonic, parse_operand, emit_instruction (instructions.asm)
 ;   expand_macro, capture_macro_line (macro_expansion.asm)
 ;   file_stack_init, open_input (file_stack.asm / init.asm)
diff --git a/assembler2/23/directives.asm b/assembler2/23/directives.asm
index a6818ae..a982403 100644
--- a/assembler2/23/directives.asm
+++ b/assembler2/23/directives.asm
@@ -1,8 +1,8 @@
 ; directives.asm - Directive dispatch, data directives, conditional assembly
 ;
-; Provides: swap_pc_with_save, process_directive, process_conditional_directive,
-;           emit_quoted, set_data_mode, data_parameters_loop,
-;           dir_reserve, dir_ifdef, dir_ifndef, dir_else, dir_endif
+; Provides: swap_pc_with_save, process_directive, emit_quoted, set_data_mode,
+;           data_parameters_loop, dir_reserve, dir_ifdef, dir_ifndef,
+;           dir_else, dir_endif
 ;
 ; Requires:
 ;   CURR_CHAR (asm.asm alias; backing storage in file_stack.asm)
@@ -163,39 +163,6 @@ dir_zp_alloc:
 
 ; On exit C=0 if processed; C=1 if not processed
 ;         A is not preserved
-process_conditional_directive:
-  JSR select_instruction_hash_table
-  JSR find_in_hash_instruction
-  BCS .not_conditional       ; Not found in IHASHTAB
-  ; Check for MODE_DIRECTIVE
-  LDA (TABP16),Y
-  CMP #MODE_DIRECTIVE
-  BNE .not_conditional
-  ; Extract handler address
-  INY
-  LDA (TABP16),Y
-  STA JUMP_TARGET16
-  INY
-  LDA (TABP16),Y
-  STA JUMP_TARGET16 + 1
-  ; Check if it's one of the 4 conditional directives
-  CMPI16 JUMP_TARGET16, dir_ifdef
-  BEQ .found
-  CMPI16 JUMP_TARGET16, dir_ifndef
-  BEQ .found
-  CMPI16 JUMP_TARGET16, dir_else
-  BEQ .found
-  CMPI16 JUMP_TARGET16, dir_endif
-  BEQ .found
-.not_conditional:
-  SEC                        ; Not a conditional directive
-  RTS
-.found:
-  JSR do_jump                ; Call handler via trampoline
-  CLC                        ; Processed
-  RTS
-
-
 ; Read and emit quoted ASCII
 ; On entry A contains the first character within quotes
 ; On exit A contains the current character after the closing quote
-- 
2.43.0

